server/plugin/api_test.go

```diff
@@ -0,0 +1,917 @@
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"bou.ke/monkey"
+	"github.com/Brightscout/mattermost-plugin-azure-devops/mocks"
+	"github.com/Brightscout/mattermost-plugin-azure-devops/server/constants"
+	"github.com/Brightscout/mattermost-plugin-azure-devops/server/serializers"
+	"github.com/golang/mock/gomock"
+	"github.com/gorilla/mux"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin/plugintest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type panicHandler struct {
+}
+
+func (ph panicHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	panic("bad handler")
+}
+
+func TestInitRoutes(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+	p.router = p.InitAPI()
+	p.InitRoutes()
+}
+
+func TestHandleStaticFiles(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+
+	p.router = p.InitAPI()
+	mockAPI.On("GetBundlePath").Return("/test-path", nil)
+	p.HandleStaticFiles()
+}
+
+func TestWithRecovery(t *testing.T) {
+	defer func() {
+		if x := recover(); x != nil {
+			require.Fail(t, "got panic")
+		}
+	}()
+
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	mockAPI.On("LogError", "Recovered from a panic", "url", "http://random", "error", "bad handler", "stack", mock.Anything)
+	p.SetAPI(mockAPI)
+
+	ph := panicHandler{}
+	handler := p.WithRecovery(ph)
+
+	req := httptest.NewRequest(http.MethodGet, "http://random", nil)
+	w := httptest.NewRecorder()
+	handler.ServeHTTP(w, req)
+
+	resp := w.Result()
+	if resp.Body != nil {
+		defer resp.Body.Close()
+		_, err := io.Copy(ioutil.Discard, resp.Body)
+		require.NoError(t, err)
+	}
+}
+
+func TestHandleAuthRequired(t *testing.T) {
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	p.API = mockAPI
+	for _, testCase := range []struct {
+		description string
+	}{
+		{
+			description: "test handleAuthRequired",
+		},
+	} {
+		t.Run(testCase.description, func(t *testing.T) {
+			timerHandler := func(w http.ResponseWriter, r *http.Request) {}
+
+			req := httptest.NewRequest(http.MethodPost, "/tasks", bytes.NewBufferString(`{}`))
+			req.Header.Add(constants.HeaderMattermostUserID, "mockUserID")
+
+			res := httptest.NewRecorder()
+
+			timerHandler(res, req)
+
+			resp := p.handleAuthRequired(timerHandler)
+			assert.NotNil(t, resp)
+		})
+	}
+}
+
+func TestHandleCreateTask(t *testing.T) {
+	defer monkey.UnpatchAll()
+	p := Plugin{}
+	mockAPI := &plugintest.API{}
+	mockCtrl := gomock.NewController(t)
+	mockedClient := mocks.NewMockClient(mockCtrl)
+	p.API = mockAPI
+	p.Client = mockedClient
+	for _, testCase := range []struct {
+		description        string
+		body               string
+		err                error
+		marshalError       error
+		statusCode         int
+		expectedStatusCode int
+		clientError        error
+	}{
+		{
+			description: "test CreateTask",
```

Comment:
```suggestion
			description: "CreateTask: valid fields",
```
same as previous PR comment
https://github.com/Brightscout/mattermost-plugin-azure-devops/pull/47#pullrequestreview-1096035931